pipeline_dp/budget_accounting.py

```diff
@@ -100,7 +101,205 @@ def compute_budgets(self):
+        return self._noise_standard_deviation
+
+
+@dataclass
+class MechanismSpecInternal:
+    """Stores sensitivity and weight not exposed in MechanismSpec."""
+    sensitivity: float
+    weight: float
+    mechanism_spec: MechanismSpec
+
+
+class PLDBudgetAccountant:
+    """Manages the privacy budget for privacy loss distributions.
+
+    It manages the privacy budget for the pipeline using the
+    Privacy Loss Distribution (PLD) implementation from Google's
+    dp_accounting library.
+    """
+
+    def __init__(self,
+                 total_epsilon: float,
+                 total_delta: float,
+                 pld_discretization: float = 1e-4):
+        """Constructs a PLDBudgetAccountant.
+
+        Args:
+            total_epsilon: epsilon for the entire pipeline.
+            total_delta: delta for the entire pipeline.
+            pld_discretization: `value_discretization_interval` in PLD library.
+                Smaller interval results in better accuracy, but increases running time.
+
+        Raises:
+            ValueError: Arguments are missing or out of range.
+        """
+
+        _validate_epsilon_delta(total_epsilon, total_delta)
+
+        self._total_epsilon = total_epsilon
+        self._total_delta = total_delta
+        self._mechanisms = []
+        self.minimum_noise_std = None
+        self._pld_discretization = pld_discretization
+
+    def request_budget(self,
+                       noise_kind: NoiseKind,
+                       sensitivity: float = 1,
+                       weight: float = 1) -> MechanismSpec:
+        """Request a budget.
+
+        Constructs a mechanism spec based on the parameters.
+        Adds the mechanism to the pipeline for future calculation.
+
+        Args:
+            noise_kind: The kind of noise distribution for the mechanism.
+            sensitivity: The sensitivity for the mechanism.
+            weight: The weight for the mechanism.
+
+        Returns:
+            A "lazy" mechanism spec object that doesn't contain the noise
+            standard deviation until compute_budgets is called.
+        """
+        if noise_kind == NoiseKind.GAUSSIAN and self._total_delta == 0:
+            raise AssertionError("Gaussian delta value must be greater than 0")
+        mechanism_spec = MechanismSpec(noise_kind=noise_kind)
+        mechanism_spec_internal = MechanismSpecInternal(
+            mechanism_spec=mechanism_spec,
+            sensitivity=sensitivity,
+            weight=weight)
+        self._mechanisms.append(mechanism_spec_internal)
+        return mechanism_spec
+
+    def compute_budgets(self):
+        """Computes the budget for the pipeline.
+
+        Composes the mechanisms and adjusts the amount of
+        noise based on given epsilon. Sets the noise for the
+        entire pipeline.
+        """
+        if not self._mechanisms:
+            return
+        if self._total_delta == 0:
+            standard_noise = len(self._mechanisms) / self._total_epsilon
```

Comment:
@dvadym This is the new calculation for standard Laplace (epsilon, 0)-DP. I am not sure that it handles sensitivity and weight per mechanism correctly. Thank you for taking a look!

Reply:
I think PLD library should handle the case of delta=0 (in case only laplacian mechanisms are requested), please try w/o this code and to see whether it works.

If not, we can submit this CL w/o processing of delta=0, I'll fix the library and then we can process delta=0 case with PLD.

Reply:
👋 @dvadym It fails in `privacy_loss_distribution.py` file in `def get_epsilon_for_delta(self, delta: float) -> float:` because it returns infinity value. The example for LaPlace retrieves the delta for epsilon, but then the delta is not 0, so perhaps this is not what we want https://github.com/google/differential-privacy/blob/main/python/dp_accounting/privacy_loss_distribution_basic_example.py#L40 ?

I changed to use `get_smallest_laplace_noise` function from https://github.com/google/differential-privacy/blob/main/python/dp_accounting/accountant.py#L63. I'm not sure this will work as the function takes one sensitivity overall, and we want a sensitivity per mechanism. 

Please let me know if I should remove the delta=0 cases and tests and we can change it when the PLD library has a fix. Thank you!